models/reservation.js

```diff
@@ -106,45 +63,34 @@ exports.reserve = function(driverId, latitude, longitude) {
       reservation.attributes.longitude = spotLatLong.longitude;
       reservation.attributes.hostId = spot.attributes.userId;
       reservation.attributes.spotId = spot.id;
-      return reservation;
-    })
-    .then(createNewReservation)
-    .then(function(reservation) {
-      return Spot.addReservationId(reservation.attributes.spotId, reservation.id);
+      return reservationsRef.child(reservation.id).update(reservation.attributes);
     })
     .then(function() {
-      return reservation;
+      return exports.get(reservation.id);
     });
 };
 
-/* Change status to navigating */
-exports.navigating = function(reservationId) {
-  return exports.updateStatus(reservationId, 'navigating');
+/* Change status to accepted and add to active lists */
+exports.accept = function(reservationId) {
+  return exports.updateStatus(reservationId, exports.ReservationStatusEnum.accepted)
+    .then(User.addReservationToActive);
 };
 
-/* Change status to occupied and add to active lists */
+/* Change status to occupied and start price timer */
 exports.occupy = function(reservationId) {
-  return exports.updateStatus(reservationId, 'occupied')
-    .then(setStartTime)
-    .then(addReservationToActive)
-    .then(function(oldReservation) {
-      // Return the updated resrvation
-      return exports.get(reservationId);  
-    });
+  return exports.updateStatus(reservationId, exports.ReservationStatusEnum.occupied)
+    .then(setStartTime);
 };
 
-/* Change reservation status, remove from both of the active lists */
+/* Change status to finished, remove from driver list,
```

Comment:
You're actually just removing it from the host's active list, not the driver's active list right?